docs/ci-integration/guides/gh-actions-integration.md

```diff
@@ -1,7 +1,52 @@
 
 # GitHub Actions integration
 
-Here is an example a GitHub Actions build, where we build the Earthly target `+build`.
+Here is an example a GitHub Actions build that uses the [https://github.com/earthly/actions/setup-earthly](earthly/actions/setup-earthly action).
+
+This example assumes an [Earthfile](../earthfile/earthfile.md) exists with a `+build` target:
+
+```yml
+# .github/workflows/ci.yml
+
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+      FORCE_COLOR: 1
+    steps:
+    - uses: earthly/actions/setup-earthly@v1
+      with:
+        version: v0.7.0
+    - uses: actions/checkout@v2
+    - name: Put back the git branch into git (Earthly uses it for tagging)
+      run: |
+        branch=""
+        if [ -n "$GITHUB_HEAD_REF" ]; then
+          branch="$GITHUB_HEAD_REF"
+        else
+          branch="${GITHUB_REF##*/}"
+        fi
+        git checkout -b "$branch" || true
+    - name: Docker Login
+      run: docker login --username "$DOCKERHUB_USERNAME" --password "$DOCKERHUB_TOKEN"
+    - name: Earthly version
+      run: earthly --version
+    - name: Run build
+      run: earthly --push +build
+```
+
+Alternatively, you can skip using the `earthly/actions/setup-earthly` job, and include
```

Comment:
I think this sentence flows slightly better with one less comma, but that's purely an opinion.
```suggestion
Alternatively, you can skip using the `earthly/actions/setup-earthly` job and include
```